### PR TITLE
CVE 2021 22569 on protobuf java in trilead api

### PR DESCRIPTION
**Pull request by BorisYaoA**
See [JENKINS-XXXXX](https://issues.jenkins-ci.org/browse/JENKINS-XXXXX).

### Submitter checklist

- [ ] JIRA issue is well described
- [ ] Appropriate autotests or explanation to why this change has no tests

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -84,7 +84,13 @@
...
             <groupId>org.jenkins-ci</groupId>
             <artifactId>trilead-ssh2</artifactId>
-            <version>build-217-jenkins-27</version>
+            <version>build-217-jenkins-211.vbb42cae44b18</version>
+            <exclusions>
+                <exclusion>
+                    <groupId>com.google.protobuf</groupId>
+                    <artifactId>protobuf-java</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (jtnord) [inline pom.xml:93]: you should not need to exclude the version here.  the maven dependency will be closest wins, and as you have a direct dependency on `protobuf-java` that would win.

it also means that if/when the upstream library is updated to use a newer protobuf the enforcer [`upperBounds`](https://maven.apache.org/enforcer/enforcer-rules/requireUpperBoundDeps.html) rule would let you know.

```suggestion
```

File: pom.xml [hunk 2]
```diff
@@ -97,5 +103,10 @@
                 </exclusion>
             </exclusions>
         </dependency>
+        <dependency>
+            <groupId>com.google.protobuf</groupId>
+            <artifactId>protobuf-java</artifactId>
+            <version>3.16.1</version>
```
Comment (jtnord) [inline pom.xml:109]:
```suggestion
            <!-- trilead-ssh2 ships with a library containing a known CVE causing scanner alerts, even though the code is not used.  so we update to a newer version without the vulnerability, but do not upgrade too far to maintain api compatability -->
            <version>3.16.1</version>
```

File: pom.xml [hunk 3]
```diff
@@ -84,7 +84,13 @@
         <dependency>
             <groupId>org.jenkins-ci</groupId>
             <artifactId>trilead-ssh2</artifactId>
-            <version>build-217-jenkins-27</version>
+            <version>build-217-jenkins-211.vbb42cae44b18</version>
```
Comment (jtnord) [inline pom.xml:87]: this obsoletes #57

**Review by jtnord — COMMENTED** [review]
Some inline comments.

Please create a [Jenkins issue](https://issues.jenkins.io/) describing the issue that is being solved here. BEE references are not visible to non CloudBees employees and should not be used for OSS pull requests.

**Comment by kuisathaverat** [comment]
Why not resolve this in the trilead-ssh2 module? https://github.com/jenkinsci/trilead-ssh2, Also I did not dig where the dependency came from in the `trilead-ssh2` but I bet it can be excluded and ignored

**Comment by kuisathaverat** [comment]
The issue can be resolved by bumping com.google.crypto.tink:tink:jar:1.6.1 in the trilead-ssh2 to a version that updates protobuff, so this PR here does not have much sense.

```
> mvn dependency:tree
[INFO] Scanning for projects...
[INFO] 
[INFO] --------------------< org.jenkins-ci:trilead-ssh2 >---------------------
[INFO] Building Ganymed SSH2 for Java build-217-jenkins-999999-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-dependency-plugin:3.3.0:tree (default-cli) @ trilead-ssh2 ---
[INFO] org.jenkins-ci:trilead-ssh2:jar:build-217-jenkins-999999-SNAPSHOT
[INFO] +- commons-io:commons-io:jar:2.11.0:test
[INFO] +- junit:junit:jar:4.13.2:test
[INFO] |  \- org.hamcrest:hamcrest-core:jar:1.3:test
[INFO] +- org.hamcrest:hamcrest:jar:2.2:test
[INFO] +- org.mockito:mockito-core:jar:4.6.1:test
[INFO] |  +- net.bytebuddy:byte-buddy:jar:1.12.10:test
[INFO] |  +- net.bytebuddy:byte-buddy-agent:jar:1.12.10:test
[INFO] |  \- org.objenesis:objenesis:jar:3.2:test
[INFO] +- net.i2p.crypto:eddsa:jar:0.3.0:compile
[INFO] +- org.connectbot.jbcrypt:jbcrypt:jar:1.0.0:compile
[INFO] +- com.google.crypto.tink:tink:jar:1.6.1:compile
[INFO] |  +- com.google.protobuf:protobuf-java:jar:3.14.0:compile
[INFO] |  \- com.google.code.gson:gson:jar:2.8.6:compile
[INFO] +- org.testcontainers:testcontainers:jar:1.17.3:test
[INFO] |  +- org.slf4j:slf4j-api:jar:1.7.36:test
[INFO] |  +- org.apache.commons:commons-compress:jar:1.21:test
[INFO] |  +- org.rnorth.duct-tape:duct-tape:jar:1.0.8:test
[INFO] |  |  \- org.jetbrains:annotations:jar:17.0.0:test
[INFO] |  +- com.github.docker-java:docker-java-api:jar:3.2.13:test
[INFO] |  |  \- com.fasterxml.jackson.core:jackson-annotations:jar:2.10.3:test
[INFO] |  \- com.github.docker-java:docker-java-transport-zerodep:jar:3.2.13:test
[INFO] |     +- com.github.docker-java:docker-java-transport:jar:3.2.13:test
[INFO] |     \- net.java.dev.jna:jna:jar:5.8.0:test
[INFO] \- org.slf4j:slf4j-nop:jar:1.7.36:test
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.030 s
[INFO] Finished at: 2022-07-21T20:31:04+02:00
[INFO] ------------------------------------------------------------------------
```

**Comment by kuisathaverat** [comment]
here is no new versions of https://github.com/google/tink, so the solution is to force the version of protobuf in trilead-ssh2

**Comment by kuisathaverat** [comment]
superseded by https://github.com/jenkinsci/trilead-ssh2/pull/102

**Comment by kuisathaverat** [comment]
BTW this vulnerability does not affect any Jenkins component due to a DoS attack and this library is not used to expose any service, it is only used to consume SSH services.

**Comment by kuisathaverat** [comment]
as I commented before this plugin is not the place to fix the CVE, this CVE come from a dependency trilead-ssh2 and there is where the issue was to be fixed, and It is fixed in the PR https://github.com/jenkinsci/trilead-ssh2/pull/102

**Pull request closed without merging**